src/platform/nrfconnect/shell/ChipShell.cpp

```diff
@@ -0,0 +1,197 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app/server/OnboardingCodesUtil.h>
+#include <platform/CHIPDeviceLayer.h>
+#include <support/CodeUtils.h>
+
+#include "shell/shell.h"
+
+#define CHIP_SHELL_MAX_BUFFER_SIZE 128
+
+using namespace chip::DeviceLayer;
+
+static CHIP_ERROR ChipShellProcessQRCode(const struct shell * shell, size_t argc, char ** argv)
```

Comment:
According to the `shell_cmd_handler` definition the command handlers should return int as follows:
```
 * @retval 0 Successful command execution.
 * @retval 1 Help printed and command not executed.
 * @retval -EINVAL Argument validation failed.
 * @retval -ENOEXEC Command not executed.
```
Should we return one of those on error?

Reply:
I aligned to to return 0 on success and -ENOEXEC on failure.